### PR TITLE
:+1: generate sub local scope option types

### LINKED ISSUE #267: option: "local to window" options has `getBuffer()` method
> * Users may access options in an unintended way
>   
>   * For example, `winfixwidth` is meant to be window-local, but users can still access it through the `getBuffer` method.

1. `getBuffer()` method exists even though `winfixwidth` etc are "local to window":
   This is an existing problem, [it was there before this PR](https://github.com/vim-denops/deno-denops-std/blob/v7.1.1/option/_generated.ts#L16462).

_Originally posted by @Milly in https://github.com/vim-denops/deno-denops-std/issues/266#issuecomment-2314621854_

### PR DESCRIPTION
**Pull request by Milly**
Fixes #267
Fixes some #268

`getTabPage()` or `setTabPage()` is not implemented.
See https://github.com/vim-denops/deno-denops-std/issues/268#issuecomment-2316587922

## Summary by CodeRabbit

- **New Features**
	- Enhanced option processing and exporting with a new `Context` type for better management of option types and constructors.
	- Improved error handling in the parsing functions for more informative logging.
	- Introduced new interfaces for local options, allowing for more specific option management across different contexts (buffer, tab page, window).

- **Bug Fixes**
	- Updated return types in parsing functions to ensure more predictable outputs and prevent runtime errors.

- **Documentation**
	- Updated comments to clarify new requirements and expectations for option handling.

- **Refactor**
	- Streamlined type checks and option declarations for improved readability and maintainability.

### CODE REVIEW

**Comment by Milly** [comment]
Blocked by #269.
```
error: TS2339 [ERROR]: Property 'getBuffer' does not exist on type 'WindowLocalOption<string>'.
      op.foldmethod.getBuffer(denops, bufnr),
                    ~~~~~~~~~
    at file:///deno-denops-std/buffer/buffer.ts:526:21
```

**Comment by Milly** [comment]
@coderabbitai review

File: option/types.ts
Comment (lambdalisue) [reply option/types.ts:105]: Not **must** but I think we can implement that with some workaround
https://github.com/vim-denops/deno-denops-std/issues/268#issuecomment-2334327049

Comment (lambdalisue) [reply option/types.ts:105]: > I think we can implement that with some workaround

No. So this should be OK

**Review by lambdalisue — APPROVED** [review]
LGTM. Can we merge this (if so, plz merge it)? @Milly

**Pull request merged**
